changelogs/unreleased/gh-9644-verbose-uri-error.md

```diff
@@ -2,3 +2,5 @@
 
 * Added a subject URI to the error that is thrown on configuration
   verification (gh-9644).
+* Added a warning with the skipped (unsuitable) URI to replicaset
+  and sharding configuration (gh-9644).
```

Comment:
Nit: The commit message of the first commit closes the issue, while it should be marked with the part-of tag. The second commit message doesn't mention the issue at all, but it should be marked as closing the issue.

Reply:
Oh, missed that with all rebases.

Reply:
Fixed.